internal/controller/apiextensions/claim/reconciler.go

```diff
@@ -528,8 +529,12 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{Requeue: true}, errors.Wrap(r.client.Status().Update(ctx, cm), errUpdateClaimStatus)
 	}
 
-	log.Debug("Successfully applied composite resource")
-	record.Event(cm, event.Normal(reasonCompositeConfigure, "Successfully applied composite resource"))
+	if oldRV != cp.GetResourceVersion() {
```

Comment:
not sure I am a fan of this construct. Am open for better patterns.

See my comments about the available applicators in Slack.

Reply:
Perhaps something like:

```go
err := r.client.Apply(ctx, cp, resource.AllowUpdateIf(func(current, desired) bool { !cmp.Equal(current, desired) }))
switch {
case resource.IsNotAllowed(err):
	log.Debug("Skipped no-op composite resource apply")
case err != nil:
	// Current error handling block
default:
	log.Debug("Successfully applied composite resource")
	record.Event(cm, event.Normal(reasonCompositeConfigure, "Successfully applied composite resource"))
}
```

I believe this would disallow the apply (i.e. patch) entirely if this reconcile didn't actually change `cp`. Using `cmp.Equal` on the entire `*Unstructured` feels inelegant but this approach at least seems syntactically more pleasant to me.

https://pkg.go.dev/github.com/crossplane/crossplane-runtime@v0.19.2/pkg/resource#AllowUpdateIf

Reply:
Was wondering whether this can still lead to multiple events when the informers are lagging behind. Just because an update is attempted does not mean an actual update happened.

Reply:
> Was wondering whether this can still lead to multiple events when the informers are lagging behind.

FWIW I _think_ the controller-runtime machinery automatically dedupes multiple events in the queue to result in one reconcile call.

Reply:
and in any case this will eventually settle. If two or more, but few events are creating on rare races, that's fine. Just cosmetics (different to the RV topic in the applicator which is about consistency).

Reply:
updated.